[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave devi…

juhaki opened the pull request:
…ce for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+	simulator {
+	}
+
+preferences {
```
nit: spacing

juhaki replied:
Thanks. I will add the spacing to "	preferences {"

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+/**
+ *  Copyright 2018 SmartThings
```
nit: copyright date

juhaki replied:
Thanks. I will update to "Copyright 2020 SmartThings".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+
+def configure() {
+	log.info "configure called"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```
Please remove the `offlinePingable` here. This does cause some load for device health and we prefer to restrict it to life safety devices.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
```
this manufacturer-specific handling (here and in `configure`) is no longer necessary now that this info is encapsulated in the `zwaveInfo` object

juhaki replied:
Ok. Thanks. This function will be delete.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+def updated() {
+	log.info "updated called"
+	configure()
+}
```
after a device is installed, `installed`, `updated`, and `configure` are all called in that order. You'd be calling `configure` three times in a row here

greens replied:
`updated` is also called when a user adjusts preferences, so whatever solution you end up with here, you'll want to make sure that it still results in the configuration sets being sent if the values have changed

juhaki replied:
Ok. thanks. I will patch this part by your comment.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+
+private secure(cmd) {
+	log.info "secure called"
+	if(zwaveInfo.zw.endsWith("s")) {
```
this (due to changes with S2) should now be `contains` instead of `endsWith`

juhaki replied:
Thanks. This function ( secure ) will be patch.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+	delayBetween([
+		secureEncap(zwave.basicV1.basicSet(value: value), endpoint),
+		"delay 500"
+	], 500)
```
I'm guessing there's meant to be a get here?

Also, it seems unnecessary to put an explicit delay command here when you're calling `delayBetween`

juhaki replied:
Thanks. I will remove the "delayBetween".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+			addChildDevice("Child Dawon Z-wave Wall Smart Switch Health", childDni, device.getHub().getId(), [
+				completedSetup: true,
+				label         : componentLabel,
+				isComponent   : false
```
Is there a reason you've chosen not to add these device as components? You'd have to create new metadata, but right now all the children would show up as separate devices on the device list. I'm not sure how this device itself is physically configured, but I believe we generally want to treat physically integrated child devices as components.

juhaki replied:
This part have intended by main component & child component.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+		def value = cmd.event== 0x03? "on" : "off"
+		endpoint ? result = changeSwitch(endpoint, value) : []
+	} else if (cmd.notificationType == 0x04) {
+		def map = [name: "tamper"]
```
you have not declared the `"Tamper"` capability.

juhaki replied:
I also think this part not needed in now. So I will remove that.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+		if (cmd.event == 0x02 || cmd.event == 0x04) {
+			map.value = detected
+		} else if (cmd.event == 0x06) {
+			map.value = clear
```
I believe you meant to make these strings

juhaki replied:
Thanks. I will patch.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
```
This can be omitted. This is also now included in the `zwaveInfo` object

juhaki replied:
Ok. Thanks. I will remove this part.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+ * */
+def ping() {
+	log.info "ping called"
+	refresh()
```
you have not defined either an argument-less `refresh` or specified a default argument in your `refresh`

You could do `def refresh(endpoint=null)`

juhaki replied:
Ok. thanks. I will patch that.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+			break
+	}
+	log.debug "ConfigurationReport event: ${map}"
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
```
Perhaps this should just be a debug log message?

juhaki replied:
This part will be remove.

juhaki replied:
This part is just log. So. I will delete this part.

greens replied:
Just make it a log message, then instead of creating this event.

juhaki replied:
Ok. I will update this line to log message.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+
+def isDawonWallSmartSwitch1() {
+	return zwaveInfo.prod.equals("0061")
+}
```
I think these could be unified into one function, preferably following the pattern we use here:

https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy#L233

juhaki replied:
Ok. This type functions will reduce to the one function.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,392 @@
+		0x84: 1,  // WakeUp
+		0x98: 1,  // Security
+		0x9C: 1,  // Sensor Alarm
+		0x71: 3   // Notification
```
please remove command classes that are not being used in this DTH

juhaki replied:
Ok. I will patch.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki reviewed (commented):
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.
DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

Please review this patch.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+	}
+
+	simulator {
+	}
```
because this is empty, it can be omitted

juhaki replied:
Ok. I will remove this code "simulator" part.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+		}
+	} catch (e) {
+		log.warn e
+	}
```
this can all be omitted.

juhaki replied:
Ok. I will patch this code.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+	initialize()
+}
+
+def updated() {
```
Like I mentioned before, when you change a preference, `updated` is called. The way you have this set up, the new preference values will not be sent to the device when they are changed by the user.

juhaki replied:
Is your mention that below method?

==> 
def updated() {
	log.info "updated called"
	configure()
}

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+def configure() {
+	log.info "configure called"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	def msrdata = getDataValue("MSR")
```
remove

juhaki replied:
Ok. I will remove this line. "def msrdata = getDataValue("MSR")"

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+
+	if (msrdata == null) {
+		commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet()
+	}
```
remove, unused/unneeded

juhaki replied:
Ok. I will remove this part. 

	if (msrdata == null) {
		commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet()
	}

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+		0x70: 2,  // Configuration
+		0x98: 1,  // Security
+		0x9C: 1,  // Sensor Alarm
+		0x71: 3   // Notification
```
This DTH does not use `Meter` or `Sensor Alarm`

juhaki replied:
Thanks. I will remove "Meter & Sensor Alarm".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+	refresh(endpoint)
+}
+
+def refresh(endpoint) {
```
you can still call this method without passing an argument from the Classic tiles, so I would recommend setting a default value of `null` for `endpoint`.

juhaki replied:
Thanks. I will add to default value to endpoint parameter.
==>
def refresh(endpoint = null) {

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+	if (switchId != null) sendHubCommand refresh(switchId)
+}
+
+private isDawonWallSmartSwitch() {
```
since the purpose of this function has changed, its name should also change. Perhaps `getNumberOfChildrenFromModel` or something like that.

juhaki replied:
Ok. I will update function name to "getNumberOfChildFromModel".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,371 @@
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+	secure(encap(cmd, endpoint))
+}
```
I think a number of these functions could be consolidated with little effort.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki reviewed (commented):
[WWST-4872, WWST-4876, WWST-4892] DawonDNS child in-wall switch Z-wave device.
DawonDNS child in-wall switch Z-wave device for Korea.
This file for Z-wave child switch items.

Please review this pr.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+
+def initialize() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
```
this function could just be inlined in `installed()`

juhaki replied:
Ok. this part will move to the installed().

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+	log.info "zwaveEvent SecurityMessageEncapsulation called: ${cmd}, ${endpoint}"
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		waveEvent(encapsulatedCommand)
```
typo: `zwaveEvent`

juhaki replied:
Thanks. I will update.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
```
This will overwrite the return value of the unencapsulated command handler.

juhaki replied:
Ok. I will remove this line.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+ * */
+def ping(endpoint = null) {
+	log.info "ping called"
+	refresh(endpoint)
```
you can just call `refresh()` without any arguments here

juhaki replied:
Ok. I will update this function by arguments type.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		def type = getNumberOfChildFromModel()
```
nit: `type` is not a well-named variable here

juhaki replied:
This part will update from "type" to "numberOfChild".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+
+	def CmdEncap = []
```
nit: variables should start with lowercase letters

juhaki replied:
This variables will update from "CmdEncap" to "cmdEncap".

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
```
unused

greens replied:
and 'command' is undefined

juhaki replied:
Ok. I will remove this part.

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}/$endpoint"
```
modern convention is to use `:` to separate endpoints in DNIs

juhaki replied:
Thanks. I will update this line from "/" to ":".

String childDni = "${device.deviceNetworkId}/$endpoint"
==> String childDni = "${device.deviceNetworkId}:$endpoint"

greens commented on devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy:
```diff
@@ -0,0 +1,346 @@
+			String childDni = "${device.deviceNetworkId}/$endpoint"
+			def componentLabel = "Dawon Smart Switch${endpoint}"
+			def child = addChildDevice("Child Switch", childDni, device.hubId,
+				[completedSetup: true, label: componentLabel, isComponent: false])
```
Again: Is there a reason you want all the devices to be split up into separate entries on the device list? They could all be integrated into one page, but you would need to create a unique ui-metadata.

juhaki replied:
This code is intended by DawonDNS multicomponent z-wave in-wall switch device.

juhaki committed:
[WWST-4872, WWST-4876, WWST-4892] DawonDNS in-wall switch Z-wave device for main items.

DawonDNS in-wall switch Z-wave device for Korea.
This file for main(Temp/Humidity Sensor) items.

juhaki commented:
I will renew gerrit items.
Move to "https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/21916/files"
